packages/server-wallet/src/wallet/index.ts

```diff
@@ -451,13 +518,35 @@ export class Wallet implements WalletInterface, ChainEventSubscriberInterface {
     const outbox: Outgoing[] = [];
     const channelResults: ChannelResult[] = [];
     let error: Error | undefined = undefined;
-    while (channels.length && !error) {
-      await this.store.lockApp(channels[0], async tx => {
+
+    // FIXME: Only get objectives which are:
+    // 1. Approved but not executed yet
+    // 2. Related to one of the channels
+    const objectives = Object.values(this.store.objectives).filter(
+      objective =>
+        // Only supports these two
+        (objective.type === 'OpenChannel' || objective.type === 'CloseChannel') &&
+        // Only runs on those with relevant channels
+        _.includes(channels, objective.data.targetChannelId) &&
+        // Only runs on pending or approved
+        (objective.status === 'approved' || // Need approved b.c. next action to take
+          objective.status === 'pending') /* Need pending because you want new channel result */
+    );
```

Comment:
This is the bit of code which asks the store "which objectives in the store are relevant for the run loop?"